### PR TITLE
Add issue templates

### LINKED ISSUE #5: Add Repository templates
Create the following templates:
- [x] Issue - Bug
- [x] Issue - Question
- [x] Issue - (New) Feature Request
- [x] Pull Request (For contributions)

Proposed changes in #12 

Additional items to complete:
- [ ] Accept philschatz/project-bot installation (enables new issues to be auto-assigned to projects)
- [ ] Review proposed templates for accepted verbiage.

### PR DESCRIPTION
**Pull request by SheepReaper**
Closes #5 

Adds the following issue templates:
- [x] Bug report
- [x] Feature Request
- [x] Question
- [x] Pull request

Additional items to complete:
- [x] Accept philschatz/project-bot installation (enables new issues to be auto-assigned to projects)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+name: Bug report
```
Comment (pierre-labastie) [inline .github/ISSUE_TEMPLATE/bug_report.md:2]: For this part, I did not know whether I should modify part or all of this. I added a title inside the single quotes on the 'Title' line, but it is redundant with the Title of the issue

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+about: Create a report to help us improve
+title: ''
+labels: bug
+assignees: ''
+
+---
+
+**Describe the bug**
```
Comment (pierre-labastie) [inline .github/ISSUE_TEMPLATE/bug_report.md:10]: Why not use headings:
```suggestion
#### Describe the bug
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+A clear and concise description of what the bug is.
+
+**To Reproduce**
+Steps to reproduce the behavior:
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. See error
+
```
Comment (pierre-labastie) [inline .github/ISSUE_TEMPLATE/bug_report.md:19]: Missing "Observed behavior"?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots/Terminal output**
+If applicable, add screenshots to help explain your problem.
+
+**Environment Details (please complete the following information):**
+ - Host OS: [e.g. output from uname -a]
```
Comment (pierre-labastie) [inline .github/ISSUE_TEMPLATE/bug_report.md:30]: Should we leave the square brackets? Also I had a hard time preventing a #<span/>1 in the uname ouput from linking to the #1 issue. Had to write `#<span/>1`. Maybe there are easier ways, but I do not want it to appear as code.

**Review by pierre-labastie — APPROVED** [review]
I've tried the bug template. I had a couple of questions (hadn't read "contributing"), which I have made in comments below. Approving anyway. We still can make changes later

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (SheepReaper) [reply .github/ISSUE_TEMPLATE/bug_report.md:2]: name: is the name of the issue template as it appears to the user after clicking the Issues tab: [ref this](https://help.github.com/en/articles/about-issue-and-pull-request-templates)
title: is both the placeholder text and default issue title when the user is filling out the issue form.
title is mutable by the user, name is not.
title is optional, and I really didn't know what a god default issue title might be. I figure using a square bracket tag might help with sorting, but the user can always just delete it when they create the issue.

label: is the important bit, this is the label that gets auto-assigned to the new issue and the is the hint the sorter uses to route the new issue to the correct project. The questions project is a bit overkill, but having it means that the actual bug tracker is less-likely to get polluted with questions. (although we could just manually curate this, i have no idea what volume to expect)

**Comment by SheepReaper** [comment]
I haven't made any additional modifications since last week, however, I did fork the repo and merge this request into my master just so that we could see the issue templates in action. Take a peek here: https://github.com/bryan5989/jhalfs/issues/new/choose (You normally get to this page by hitting the new issue button)
GitHub issue templates are weird, they have a yaml header to tell GitHub some metadata but the actual template text is markdown. (and the bug and pull templates have to have specific names, but all others can be whatever names)
Hopefully this illustrates how the template translates into the form the user fills.
@pierre-labastie @jhuntwork

**Comment by pierre-labastie** [comment]
OK, I understand now: once the files are in master, they'll appear each time you open an issue, but without the headers (containing name:, etc). That seems OK to me. I still think that "observed behavior" should be added after "steps to reproduce". But depending on the bug, I agree that there is not always a clear separation between the steps and the observation...

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 3 above)
Comment (SheepReaper) [reply .github/ISSUE_TEMPLATE/bug_report.md:19]: Fixed in https://github.com/automate-lfs/jhalfs/pull/12/commits/268a3ed847a70d92b248f1a436d5c14452231de0

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (SheepReaper) [reply .github/ISSUE_TEMPLATE/bug_report.md:10]: ** surrounding content and followed by newline is rendered as a heading in issue templates. (doesn't look like it if you directly view the template in markdown in file view, but it renders different when actually creating an issue.)

**Comment by SheepReaper** [comment]
I've finished making all the edits I think needed making, and work with the automation and notifications. I think this is ready to squash + merge.

**Comment by pierre-labastie** [comment]
Seems OK to me. Can you do the squash + merge? (we'll have to discuss our policy about who's supposed to merge to master)...

**Pull request merged**